File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/musicbrainz/artist/get/ArtistGetReleaseGroup.java
package uk.co.ourfriendirony.medianotifier.clients.musicbrainz.artist.get;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import uk.co.ourfriendirony.medianotifier.general.MultiDateDeserializer;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "id",
        "disambiguation",
        "title",
        "secondary-type-ids",
        "primary-type",
        "secondary-types",
        "first-release-date",
        "primary-type-id"
})
public class ArtistGetReleaseGroup {

    @JsonProperty("id")
    private String id;
    @JsonProperty("disambiguation")
    private String disambiguation;
    @JsonProperty("title")
    private String title;
    @JsonProperty("secondary-type-ids")
    private List<String> secondaryTypeIds = null;
    @JsonProperty("primary-type")
    private String primaryType;
    @JsonProperty("secondary-types")
    private List<String> secondaryTypes = null;
    @JsonProperty("first-release-date")
    @JsonDeserialize(using = MultiDateDeserializer.class)
    private Date firstReleaseDate;
    @JsonProperty("primary-type-id")
    private String primaryTypeId;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("id")
    public String getId() {
        return id;
    }

    @JsonProperty("id")
    public void setId(String id) {
        this.id = id;
    }

    @JsonProperty("disambiguation")
    public String getDisambiguation() {
        return disambiguation;
    }

    @JsonProperty("disambiguation")
    public void setDisambiguation(String disambiguation) {
        this.disambiguation = disambiguation;
    }

    @JsonProperty("title")
    public String getTitle() {
        return title;
    }

    @JsonProperty("title")
    public void setTitle(String title) {
        this.title = title;
    }

    @JsonProperty("secondary-type-ids")
    public List<String> getSecondaryTypeIds() {
        return secondaryTypeIds;
    }

    @JsonProperty("secondary-type-ids")
    public void setSecondaryTypeIds(List<String> secondaryTypeIds) {
        this.secondaryTypeIds = secondaryTypeIds;
    }

    @JsonProperty("primary-type")
    public String getPrimaryType() {
        return primaryType;
    }

    @JsonProperty("primary-type")
    public void setPrimaryType(String primaryType) {
        this.primaryType = primaryType;
    }

    @JsonProperty("secondary-types")
    public List<String> getSecondaryTypes() {
        return secondaryTypes;
    }

    @JsonProperty("secondary-types")
    public void setSecondaryTypes(List<String> secondaryTypes) {
        this.secondaryTypes = secondaryTypes;
    }

    @JsonProperty("first-release-date")
    public Date getFirstReleaseDate() {
        return firstReleaseDate;
    }

    @JsonProperty("first-release-date")
    public void setFirstReleaseDate(Date firstReleaseDate) {
        this.firstReleaseDate = firstReleaseDate;
    }

    @JsonProperty("primary-type-id")
    public String getPrimaryTypeId() {
        return primaryTypeId;
    }

    @JsonProperty("primary-type-id")
    public void setPrimaryTypeId(String primaryTypeId) {
        this.primaryTypeId = primaryTypeId;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/kotlin/uk/co/ourfriendirony/medianotifier/clients/musicbrainz/artist/get/ArtistGetReleaseGroup.kt
package uk.co.ourfriendirony.medianotifier.clients.musicbrainz.artist.get

import com.fasterxml.jackson.annotation.*
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import uk.co.ourfriendirony.medianotifier.general.MultiDateDeserializer
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonPropertyOrder(
    "id",
    "disambiguation",
    "title",
    "secondary-type-ids",
    "primary-type",
    "secondary-types",
    "first-release-date",
    "primary-type-id"
)
class ArtistGetReleaseGroup {
    @get:JsonProperty("id")
    @set:JsonProperty("id")
    @JsonProperty("id")
    var id: String? = null

    @get:JsonProperty("disambiguation")
    @set:JsonProperty("disambiguation")
    @JsonProperty("disambiguation")
    var disambiguation: String? = null

    @get:JsonProperty("title")
    @set:JsonProperty("title")
    @JsonProperty("title")
    var title: String? = null

    @get:JsonProperty("secondary-type-ids")
    @set:JsonProperty("secondary-type-ids")
    @JsonProperty("secondary-type-ids")
    var secondaryTypeIds: List<String>? = null

    @get:JsonProperty("primary-type")
    @set:JsonProperty("primary-type")
    @JsonProperty("primary-type")
    var primaryType: String? = null

    @get:JsonProperty("secondary-types")
    @set:JsonProperty("secondary-types")
    @JsonProperty("secondary-types")
    var secondaryTypes: List<String>? = null

    @get:JsonProperty("first-release-date")
    @set:JsonProperty("first-release-date")
    @JsonProperty("first-release-date")
    @JsonDeserialize(using = MultiDateDeserializer::class)
    var firstReleaseDate: Date? = null

    @get:JsonProperty("primary-type-id")
    @set:JsonProperty("primary-type-id")
    @JsonProperty("primary-type-id")
    var primaryTypeId: String? = null
}